torchmetrics/image/map.py

```diff
@@ -0,0 +1,364 @@
+COCO_STATS_MAR_VALUE_INDEX = 8
+
+
+@dataclass
+class MAPMetricResults:
+    """Dataclass to wrap the final mAP results."""
+
+    map_value: Tensor
+    mar_value: Tensor
+    map_per_class_value: List[Tensor]
+    mar_per_class_value: List[Tensor]
+
+
+class _hide_prints:
+    """Internal helper context to suppress the default output of the pycocotools package."""
+
+    def __init__(self) -> None:
+        self._original_stdout = None
+
+    def __enter__(self) -> None:
+        self._original_stdout = sys.stdout
+        sys.stdout = open(os.devnull, "w")
+
+    def __exit__(self, exc_type, exc_val, exc_tb) -> None:  # type: ignore
+        sys.stdout.close()
+        sys.stdout = self._original_stdout
+
+
+def _input_validator(preds: List[Dict[str, torch.Tensor]], targets: List[Dict[str, torch.Tensor]]) -> None:
+    """Ensure the correct input format of `preds` and `targets`"""
+
+    if not isinstance(preds, list):
+        raise ValueError("Expected argument `preds` to be of type List")
+    if not isinstance(targets, list):
+        raise ValueError("Expected argument `target` to be of type List")
+    if len(preds) != len(targets):
+        raise ValueError("Expected argument `preds` and `target` to have the same length")
+
+    for k in ["detection_boxes", "detection_scores", "detection_classes"]:
+        if any(k not in p for p in preds):
+            raise ValueError(f"Expected all dicts in `preds` to contain the `{k}` key")
+
+    for k in ["groundtruth_boxes", "groundtruth_classes"]:
+        if any(k not in p for p in targets):
+            raise ValueError(f"Expected all dicts in `target` to contain the `{k}` key")
+
+    if any(type(pred["detection_boxes"]) is not torch.Tensor for pred in preds):
+        raise ValueError("Expected all detection_boxes in `preds` to be of type torch.Tensor")
+    if any(type(pred["detection_scores"]) is not torch.Tensor for pred in preds):
+        raise ValueError("Expected all detection_scores in `preds` to be of type torch.Tensor")
+    if any(type(pred["detection_classes"]) is not torch.Tensor for pred in preds):
+        raise ValueError("Expected all detection_classes in `preds` to be of type torch.Tensor")
+    if any(type(target["groundtruth_boxes"]) is not torch.Tensor for target in targets):
+        raise ValueError("Expected all groundtruth_boxes in `target` to be of type torch.Tensor")
+    if any(type(target["groundtruth_classes"]) is not torch.Tensor for target in targets):
+        raise ValueError("Expected all groundtruth_classes in `target` to be of type torch.Tensor")
+
+
+class MAP(Metric):
+    r"""
+    Computes the `Mean-Average-Precision (mAP) and Mean-Average-Recall (mAR)`_ for object detection predictions.
+    Optionally, the mAP and mAR values can be calculated per class.
+
+    Predicted boxes and targets have to be in COCO format with the box score at the end
+    (x-top left, y-top left, width, height, score). See the `update` function for more information
```

Comment:
It seems that this description is outdated.

Reply:
Thanks for the finding. Fixed it.